### ISSUE TITLE
Error on class 'Illuminate\Cache\FileStore' does not have a method 'tags'

### ISSUE THREAD

**Issue by minkbear** [Zizaco/entrust#422]
I face to the error below.

```php
call_user_func_array() expects parameter 1 to be a valid callback, class 'Illuminate\Cache\FileStore' does not have a method 'tags'
```

**Root cause**

```
Note: Cache tags are not supported when using the file or database cache drivers.
```
ref: http://laravel.com/docs/5.1/cache#cache-tags

**Solution**
Change from
```
CACHE_DRIVER=file
```
to
```
CACHE_DRIVER=array
```

and I suggest to add into README

**Comment by imamassi** [comment]
the solution worked! But I don't think that it has to be changed... Hope it fixed soon, or well add the solution to the readme I think.

**Comment by undermink** [comment]
yes works fine:)

**Comment by alkzzz** [comment]
Having this kind of error too. Wondering is it okay to leave the CACHE_DRIVER = array as a configuration?

**Comment by MightySCollins** [comment]
This seems to fix the error for me but I now have it doing 56 queries to load some pages.
From what I can tell it is not caching properly on array and also with cache set to APC

This is really annoying as it add to the page load time

**Comment by rephiscorth** [comment]
is changing the cache to `array` the actual solution or is it a temporary workaround?

**Comment by MightySCollins** [comment]
It seems to be a temporary workaround as for me it adds queries

**Comment by cryode** [comment]
While not an Entrust user, I can tell you that tags are only supported by some of Laravel's cache drivers. [They are not supported by the File and Database cache drivers.](http://laravel.com/docs/5.1/cache#cache-tags) Using something like Memcached for cache would be a recommended solution. The Array driver is not an actual caching mechanism, since it does not persist any data outside of the current request.

In an ideal world, Entrust would be compatible with all Laravel cache drivers. However, if it's using tags under the hood, developers will be required to use a cache driver that supports tags.

**Comment by acabin** [comment]
After read Entrust's code, I think there is not necessary for using tags.

If one role (or user) has changed, others can hold their caches, but not flush all roles.

**Comment by MightySCollins** [comment]
I agree like @cryode said although it might be better with tags users should not need to change what they are currently using and also alot of developers like to use the file driver whilst working locally.

**Comment by acabin** [comment]
@MightySCollins I think #445 has fixed the issue, waiting for merge.

**Comment by superdummy** [comment]
+1  for removing the tagging Dependencies.  I am running in the same issues (using the File Cache Driver)...

**Comment by case90** [comment]
I keeping the same error, even after change .env file in the CACHE_DRIVER=file to CACHE_DRIVER=array

**Comment by harshadvala** [comment]
use "zizaco/entrust": "dev-laravel-5" instead of "zizaco/entrust": "5.2.x-dev"

**Comment by heriagape** [comment]
@harshadvala Thank you very much, finally managed to work

**Comment by omarjebari** [comment]
It doesn't make sense to have to switch drivers. Thanks for the fix.

**Comment by dariopad** [comment]
anyway memcached is the best solution for Cache tagging.

**Comment by jonjie0317** [comment]
@harshadvala is it safe or fine to use `dev-laravel-5` instead of using `5.2.x-dev`  as per the documentation?
